jupyterhub/app.py

```diff
@@ -544,6 +544,20 @@ def _authenticator_default(self):
         """
     ).tag(config=True)
 
+    concurrent_spawn_limit = Integer(
+        None,
+        allow_none=True,
+        help="""
+        Maximum number of concurrent users that can be spawning at a time.
+
+        If more than this many users attempt to spawn at a time, they
+        enter an exponential delay with a timeout.
+
+        If set to `None`, no concurrent limit is enforced. If set to 0,
+        then no spawning is allowed.
```

Comment:
A Hub with no spawning allowed doesn't make sense (current 429 try again later would not be appropriate in this case). limit=0 should be no limit (and the default; I'd like to avoid `allow_none=True` on number traits in general).